src/wp-includes/class-wp-theme.php

```diff
@@ -1460,6 +1460,64 @@ public function is_allowed( $check = 'both', $blog_id = null ) {
 		return false;
 	}
 
+	/**
+	 * Returns whether this theme is a block-based theme or not.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @return bool
+	 */
+	public function is_block_based() {
+		$paths_to_index_block_template = array(
+			$this->get_file_path( '/block-templates/index.html' ),
+			$this->get_file_path( '/templates/index.html' ),
+		);
+
+		foreach ( $paths_to_index_block_template as $path_to_index_block_template ) {
+			if ( is_file( $path_to_index_block_template ) && is_readable( $path_to_index_block_template ) ) {
```

Comment:
`is_file()` also checks (like `is_readable()`) that the file name given exists AND that it's a "regular file." As this is a HTML file, the check of "regular file" seems unnecessary. Though not critical, it could confuse contributors as to why it's needed.

@anton-vlasenko do you foresee the possibility of the HTML file not being a regular file?

Reply:
@hellofromtonya 
First of all, thank you for the review.
The reason to use `is_file` is that someone might create a directory named `index.html`.
Of course, this is unlikely to happen.
But, if we are expecting a file, I thought it logical to check that we are dealing with a file (not a directory).
Please feel free to remove the `is_file( $path_to_index_block_template ) &&` check if you think it's redundant.

Reply:
Huh, you're right. And even though there's a low chance, it's not impossible or completely unlikely it could happen. Thanks for pointing that out. Yes, let's keep it in just in case to guard against that possibility.